### DISCUSSION TITLE
Stage selected line(s) and Reset selected lines(s) missing from context menu in Commit

### DISCUSSION THREAD

**Discussion by slemery1965** [gitextensions/gitextensions#9408] [Q&A]
Does anyone know what drives whether or not the 'Stage selected line(s)' and 'Reset selected line(s)' appears in the diff window's context menu in the Commit interface.  As you can see here, I am missing these items from my context menu in this repo, but they used to be there!

![image](https://user-images.githubusercontent.com/87833738/126711958-bf95ce48-8228-40ab-a1d4-8bfb1a163eb4.png)

This seems to be a repo specific issue for me because the these options are available in the context menu in other repos I'm accessing but for some reason are no longer available in this repo.

Any ideas as to what might have gotten updated in my repo that is preventing this options from being available to me?

Thanks, Sue.

**Comment by gerhardol** [comment]
What version?

In 3.5, FileViewer.cs
```c#
            SupportLinePatching = ((IsDiffView(_viewMode) && (text?.Contains("@@") ?? false))
                        || (item is not null && item.Item.IsNew && (item.Item.Staged is StagedStatus.WorkTree or StagedStatus.Index)))
                    && !Module.IsBareRepository()
                    && File.Exists(_fullPathResolver.Resolve(fileName));
```

But this should be fulfilled in the example from what I see

**Reply by slemery1965** [reply]
I also have version 3.5.  How to I get to FileViewer.cs?

**Reply by gerhardol** [reply]
Menu visibility is set here:
https://github.com/gitextensions/gitextensions/blob/master/GitUI/Editor/FileViewer.cs#L791

SupportLinePatching  is set here
https://github.com/gitextensions/gitextensions/blob/master/GitUI/Editor/FileViewer.cs#L962

Implementation changed in 3.5, it is possible to linepatch also for artificial commits.
The logic is intended to be the same.

A repo where this is visible is needed, maybe you can debug yourselves.

**Reply by slemery1965** [reply]
Thank you SO much for this info!  I've downloaded the code base and will try to debug ourselves.  We'll reach out if we have any other questions!

**Comment by vbjay** [comment]
Doesn't look like there is a selected line. Notice copy is disabled too.

**Reply by gerhardol** [reply]
No selected line (so Copy is not made visible), but stage/unstage/reset are enabled when the file is shown.
